src/components/tooltip/TooltipManager.ts

```diff
@@ -0,0 +1,199 @@
+import { TOOLTIP_DELAY_MS } from "./resources";
+
+export default class TooltipManager {
+  keyDownHandler = (event: KeyboardEvent): void => {
+    if (event.key === "Escape") {
+      const referenceEl = event.currentTarget as HTMLElement;
+      const tooltipEl = this.registeredElements.get(referenceEl);
+
+      if (tooltipEl) {
+        this.clearHoverTimeout(tooltipEl);
+        this.toggleTooltip(referenceEl, tooltipEl, false);
+      }
+    }
+  };
+
+  mouseEnterShow = (event: MouseEvent): void => {
+    this.hoverEvent(event, true);
+  };
+
+  mouseLeaveHide = (event: MouseEvent): void => {
+    this.hoverEvent(event, false);
+  };
+
+  clickHandler = (event: MouseEvent): void => {
+    const referenceEl = event.currentTarget as HTMLElement;
+    this.clickedTooltip = this.registeredElements.get(referenceEl);
+  };
+
+  focusShow = (event: FocusEvent): void => {
+    this.focusEvent(event, true);
+  };
+
+  blurHide = (event: FocusEvent): void => {
+    this.focusEvent(event, false);
+  };
+
+  hoverTimeouts: WeakMap<HTMLCalciteTooltipElement, number> = new WeakMap();
+
+  clickedTooltip: HTMLCalciteTooltipElement;
+
+  activeTooltipEl: HTMLCalciteTooltipElement;
+
+  // --------------------------------------------------------------------------
+  //
+  //  Properties
+  //
+  // --------------------------------------------------------------------------
+
+  registeredElements = new WeakMap<HTMLElement, HTMLCalciteTooltipElement>();
+
+  registerElement(referenceEl: HTMLElement, tooltip: HTMLCalciteTooltipElement): void {
+    this.registeredElements.set(referenceEl, tooltip);
+
+    if (tooltip.open) {
+      this.addOpenListeners(referenceEl);
+    } else {
+      this.addClosedListeners(referenceEl);
+    }
+  }
+
+  private addOpenListeners(referenceEl: HTMLElement): void {
+    referenceEl.addEventListener("keydown", this.keyDownHandler);
+    referenceEl.addEventListener("mouseout", this.mouseLeaveHide);
+    referenceEl.addEventListener("blur", this.blurHide);
+  }
+
+  private addClosedListeners(referenceEl: HTMLElement): void {
+    referenceEl.addEventListener("click", this.clickHandler);
+    referenceEl.addEventListener("focus", this.focusShow);
+    referenceEl.addEventListener("mouseover", this.mouseEnterShow);
+  }
+
+  unregisterElement(referenceEl: HTMLElement): void {
+    this.registeredElements.delete(referenceEl);
+
+    this.removeOpenListeners(referenceEl);
+    this.removeClosedListeners(referenceEl);
+  }
+
+  private removeClosedListeners(referenceEl: HTMLElement): void {
+    referenceEl.removeEventListener("click", this.clickHandler);
+    referenceEl.removeEventListener("focus", this.focusShow);
+    referenceEl.removeEventListener("mouseover", this.mouseEnterShow);
+  }
+
+  private removeOpenListeners(referenceEl: HTMLElement): void {
+    referenceEl.removeEventListener("keydown", this.keyDownHandler);
+    referenceEl.removeEventListener("mouseout", this.mouseLeaveHide);
```

Comment:
I think we need to add an event listener to the tooltip mouseout too to close it if the tooltip is hovered off.